homeassistant/components/thermostat/radiotherm.py

```diff
@@ -0,0 +1,127 @@
+"""
+homeassistant.components.thermostat.radiotherm
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Adds support for Radio Thermostat wifi-enabled home thermostats
+"""
+import logging
+import datetime
+from urllib.error import URLError
+
+from homeassistant.components.thermostat import (ThermostatDevice, STATE_COOL,
+                                                 STATE_IDLE, STATE_HEAT)
+from homeassistant.const import (CONF_HOST, CONF_NAME, TEMP_FAHRENHEIT)
+
+REQUIREMENTS = ['radiotherm==1.2']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the Radio Thermostat. """
+    logger = logging.getLogger(__name__)
+
+    try:
+        import radiotherm
+    except ImportError:
+        logger.exception(
+            "Error while importing dependency radiotherm. "
+            "Did you maybe not install the radiotherm dependency?")
+        return
+
+    # Detect hosts with hass discovery, config or radiotherm discovery
+    hosts = []
+    if discovery_info:
+        logger.info('hass radiotherm discovery', discovery_info)
+    elif CONF_HOST in config:
+        hosts = [config[CONF_HOST]]
+    else:
+        hosts.append(radiotherm.discover.discover_address())
+
+    name = config.get(CONF_NAME)
+    if hosts is None:
+        logger.error("no radiotherm thermostats detected")
+        return
+
+    tstats = []
+
+    for host in hosts:
+        try:
+            tstat = radiotherm.get_thermostat(host)
+            tstats.append(RadioThermostat(tstat))
+        except (URLError, OSError):
+            logger.exception(
+                "Unable to connect to Radio Thermostat @{}".format(host))
+
+    add_devices(tstats)
+
+
+class RadioThermostat(ThermostatDevice):
+    """ Represent a Radio Thermostat. """
+
+    def __init__(self, device, name=None):
+        self.device = device
+        if name:
+            self.set_name(name)
+        self.set_time()
+
+    @property
+    def name(self):
+        """ Returns the name of the Radio Thermostat. """
+        return self.device.name['raw']
+
+    @property
+    def unit_of_measurement(self):
+        """ Unit of measurement this thermostat expresses itself in. """
+        return TEMP_FAHRENHEIT
+
+    @property
+    def device_state_attributes(self):
+        """ Returns device specific state attributes. """
+        # Move these to Thermostat Device and make them global
+        return {
+            "humidity": None,
+            "target_humidity": None,
+            "fan": self.device.fmode['human'],
+            "mode": self.device.tmode['human']
+        }
+
+    @property
+    def current_temperature(self):
+        """ Returns the current temperature. """
+        return self.device.temp['raw']
```

Comment:
Is this querying the device? If that is the case, please implement an `update` method that caches the needed values.